src/components/transport_manager/src/cloud/websocket_client_connection.cc

```diff
@@ -141,31 +142,36 @@ TransportAdapter::Error WebsocketClientConnection::Start() {
       SDL_LOG_ERROR("Failed to add certificate authority: "
                     << cloud_properties.certificate);
       SDL_LOG_ERROR(str_err);
-      Shutdown();
+      // Reset Websocket stream
+      wss_.reset();
+      wss_ = std::make_shared<WSS>(ioc_, ctx_);
       return TransportAdapter::FAIL;
     }
 
     // Perform SSL Handshake
-    wss_.next_layer().handshake(ssl::stream_base::client, ec);
+    wss_->next_layer().handshake(ssl::stream_base::client, ec);
 
     if (ec) {
       std::string str_err = "ErrorMessage: " + ec.message();
       SDL_LOG_ERROR("Could not complete SSL Handshake failed with host/port: "
                     << host << ":" << port);
       SDL_LOG_ERROR(str_err);
-      Shutdown();
+      // Reset SSL Context and websocket stream
+      ctx_ = ssl::context(ssl::context::tlsv12_client);
```

Comment:
Seems that you'll need to apply the cipher list again here:
```
  const char* wss_ciphers =
      "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-"
      "CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-"
      "SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-"
      "AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256";
  SSL_CTX_set_cipher_list(ctx_.native_handle(), wss_ciphers);
```
Could probably create a function to clear out the connection, so that it doesn't need to be copied in several places.

Reply:
Added in https://github.com/smartdevicelink/sdl_core/pull/3891/commits/db339b798d1e706246f4ffdce4abe474d24fc018